tooling/minitest/src/tests/raw_eq.rs

```diff
@@ -1,181 +1,214 @@
 use crate::*;
 
+#[test]
+fn true_raw_eq() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.declare_function();
+    let dest = f.declare_local::<bool>();
+    let left = f.declare_local::<[u8; 2]>();
+    let right = f.declare_local::<[u8; 2]>();
+
+    let pointee = layout(size(2), align(1));
+    let ptr_ty = ref_ty(pointee);
+
+    let left_ptr = addr_of(left, ptr_ty);
+    let right_ptr = addr_of(right, ptr_ty);
+
+    f.storage_live(dest);
+    f.storage_live(left);
+    f.storage_live(right);
+
+    f.assign(left, array(&[const_int(42u8); 2], <u8>::get_type()));
+    f.assign(right, array(&[const_int(42u8); 2], <u8>::get_type()));
+
+    f.raw_eq(dest, left_ptr, right_ptr);
+
+    f.assume(eq(bool_to_int::<u8>(load(dest)), bool_to_int::<u8>(const_bool(true))));
+    f.exit();
+
+    let f = p.finish_function(f);
+    let p = p.finish_program(f);
+    assert_stop(p);
+}
+
+#[test]
+fn false_raw_eq() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.declare_function();
+    let dest = f.declare_local::<bool>();
+    let left = f.declare_local::<[u8; 2]>();
+    let right = f.declare_local::<[u8; 2]>();
+
+    let pointee = layout(size(2), align(1));
+    let ptr_ty = ref_ty(pointee);
+
+    let left_ptr = addr_of(left, ptr_ty);
+    let right_ptr = addr_of(right, ptr_ty);
+
+    f.storage_live(dest);
+    f.storage_live(left);
+    f.storage_live(right);
+
+    f.assign(left, array(&[const_int(42u8); 2], <u8>::get_type()));
+    f.assign(right, array(&[const_int(57u8); 2], <u8>::get_type()));
+
+    f.raw_eq(dest, left_ptr, right_ptr);
+
+    f.assume(eq(bool_to_int::<u8>(load(dest)), bool_to_int::<u8>(const_bool(false))));
+    f.exit();
+
+    let f = p.finish_function(f);
+    let p = p.finish_program(f);
+    assert_stop(p);
+}
+
 #[test]
 fn uninit_raw_eq() {
-    let locals = vec![bool_ty(), <[u8; 2]>::get_type(), <[u8; 2]>::get_type()];
+    let mut p = ProgramBuilder::new();
+    let mut f = p.declare_function();
+    let dest = f.declare_local::<bool>();
+    let left = f.declare_local::<[u8; 2]>();
+    let right = f.declare_local::<[u8; 2]>();
+
+    f.storage_live(dest);
+    f.storage_live(left);
+    f.storage_live(right);
 
     let pointee = layout(size(2), align(1));
     let ptr_ty = ref_ty(pointee);
 
-    let b0 = block!(
-        storage_live(0),
-        storage_live(1),
-        storage_live(2),
-        Terminator::Intrinsic {
-            intrinsic: IntrinsicOp::RawEq,
-            arguments: list!(addr_of(local(1), ptr_ty), addr_of(local(2), ptr_ty)),
-            ret: local(0),
-            next_block: Some(BbName(Name::from_internal(1)))
-        }
-    );
-    let b1 = block!(exit());
-
-    let f = function(Ret::No, 0, &locals, &[b0, b1]);
-    let p = program(&[f]);
+    let left_ptr = addr_of(left, ptr_ty);
+    let right_ptr = addr_of(right, ptr_ty);
+
+    f.raw_eq(dest, left_ptr, right_ptr);
+    f.return_();
```

Comment:
Main should `exit`, not `return`. (It makes no difference here as the code is unreachable, but it still seems better to do this consistently.)